action.yml

```diff
@@ -0,0 +1,42 @@
+name: 'Setup Dynatrace Monitoring'
+description: 'Deploys the OneAgent to the Kubernetes Cluster running Keptn. This action assumes that kubectl is available and connected to the correct cluster'
+inputs:
+  DTTenant:
+    description: Dynatrace Tenant
+    required: true
+  DTAPIToken:
+    description: Dynatrace API Token
+    required: true
+  DTPaaSToken:
+    description: Dynatrace PaaS Token
+    required: true
+  KeptnSelfMonitoringProject:
+    description: Name of the Keptn selfmonitoring project
+    required: false
+    default: 'keptn'
+  DynatraceServiceVersion:
+    description: Version of the dynatrace-service
+    default: '0.12.0'
+    required: false
+  DynatraceSLIServiceVersion:
+    description: Version of the dynatrace-sli-service
+    default: '0.9.0'
+    required: false
+runs:
+  using: composite
+  steps:
+    - shell: bash
+      id: deploy_dynatrace
+      env:
+        DT_TENANT: ${{ inputs.DTTenant }}
+        DT_API_TOKEN: ${{ inputs.DTAPIToken }}
+        DT_PAAS_TOKEN: ${{ inputs.DTPaaSToken }}
+        KEPTN_SELF_MONITORING_PROJECT: ${{ inputs.KeptnSelfMonitoringProject }}
+        BASE_PATH: ${{ github.action_path }}
+        DYNATRACE_SERVICE_VERSION: ${{ inputs.DynatraceServiceVersion }}
+        DYNATRACE_SLI_SERVICE_VERSION: ${{ inputs.DynatraceSLIServiceVersion }}
```

Comment:
we need to figure out a better way of passing secrets here. `deploy_dynatrace.sh` uses those as environment variables, but what if any script within deploy_dynatrace.sh is loaded via `wget`/`curl` and gets compromised? (Btw, this is what happened to CodeCov - see https://about.codecov.io/security-update/ )

Reply:
We talked this through, and for composite actions that call bash scripts there is currently no better solution.

The best solution would be to rewrite some parts of this Action as a NodeJS action, and use GitHub Script, same as we have it in gh-action-send-event: https://github.com/keptn/gh-action-send-event/blob/0aeea957735257246159ddab33346f47829113b3/index.js#L5-L7

I will create a separate issue for this.